### DISCUSSION TITLE
AssignmentExpression parsing issue

### DISCUSSION THREAD

**Discussion by kayabe** [dop251/goja#455] [General]
https://github.com/dop251/goja/blob/3b8a68ca89b4fa7086a4236695032e10a69b2472/parser/expression.go#L1147

why would operator `+=` be used as `+`?

**Comment by dop251** [comment]
The code looks fine to me. If you think you have found a bug, please provide a test case that fails.

**Reply by kayabe** [reply]
its not about a bug its about the case in which if use `Operator.String()` in a `AssignmentExpression` it removes `=` for `+=` so it only returns the string of `+`

**Reply by dop251** [reply]
Why is it a problem? The fact that the operator sits within an AssignmentExpression provides enough information.

**Reply by kayabe** [reply]
using the parser to turn ast back into code and using `Operator.String()` as is i would have to do `Operator.String() + "="` if operator is not just `=`
